core/src/main/java/feign/Retryer.java

```diff
@@ -47,35 +48,38 @@ public Default(long period, long maxPeriod, int maxAttempts) {
       this.attempt = 1;
     }
 
-    // visible for testing;
-    protected long currentTimeMillis() {
-      return System.currentTimeMillis();
-    }
-
+    @Override
     public void continueOrPropagate(RetryableException e) {
-      if (attempt++ >= maxAttempts) {
+      if (attempt >= maxAttempts) {
```

Comment:
This change is not backward compatible.  We count the attempt before sleeping intentionally.  This change will reduce the attempts by 1, please revert this change.